test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -1956,42 +1956,70 @@ def forward(self, input):
         x = torch.ones(5, 6)
         self.run_test(DimArange(), x)
 
+    def _test_compare_ops(self, model, num_inputs):
```

Comment:
What is the output data type in PyTorch if inputs don't have the same data type?

Reply:
The output type is always bool, regardless of the input types.

---
In reply to: [424530939](https://github.com/pytorch/pytorch/pull/38311#discussion_r424530939) [](ancestors = 424530939)